File: VB-Examples/01_MIMIE Message/Attachment/DeleteAttachment.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Email

Namespace DeleteAttachment
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			' Create an instance of MailMessage and load an email file
			Dim mail As MailMessage = MailMessage.Load("..\..\..\..\..\..\..\Data\AttachmentSample.eml")
			' Delete the attachment by index
			mail.Attachments.RemoveAt(1)
			' Delete the attachment by attachment name
			Dim i As Integer = 0
			Do While i < mail.Attachments.Count
				Dim attach As Attachment = mail.Attachments(i)
				If attach.ContentType.Name = "Sample.pdf" Then
					mail.Attachments.Remove(attach)
				End If
				i += 1
			Loop
			mail.Save("HasDeletedAttachment.eml", MailMessageFormat.Eml)
            MessageBox.Show("Completed")
		End Sub
	End Class
End Namespace

File: CS-Examples/01_MIMIE Message/Attachment/DeleteAttachment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Email;

namespace DeleteAttachment
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create an instance of MailMessage and load an email file
            MailMessage mail = MailMessage.Load(@"..\..\..\..\..\..\..\Data\AttachmentSample.eml");
            // Delete the attachment by index
            mail.Attachments.RemoveAt(1);
            // Delete the attachment by attachment name
            for (int i = 0; i < mail.Attachments.Count; i++)
            {
                Attachment attach = mail.Attachments[i];
                if (attach.ContentType.Name == "Sample.pdf")
                {
                    mail.Attachments.Remove(attach);
                }
            }
            mail.Save("HasDeletedAttachment.eml", MailMessageFormat.Eml);
            MessageBox.Show("Completed");
        }        
    }
}